### PR TITLE
RSpec Test For the App Controllers

### PR DESCRIPTION
**Pull request by zieeco**
**In this milestone, I did the following**
- Added test for the `users` controller.
- Added test for the `posts` controller.

This part of my app features adding unit testing using the RSpec test framework.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+      it 'get the contents in the user show page' do
+        expect(response.body).to include('User Details')
+      end
+
+      it 'does not render a different template' do
+        expect(response).to_not render_template(:index)
+        expect(response).to_not render_template(:new)
+      end
+    end
```
Comment (TedLivist) [inline spec/requests/users_spec.rb:42]: - [ ] Let's keep our test codes neat. Each `describe` block should be stand-alone, not nested inside another. Kindly move this `describe` block outside the other one.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      it 'get the contents in the user show page' do
+        expect(response.body).to include('Post Details')
+      end
+
+      it 'does not render a different template' do
+        expect(response).to_not render_template(:index)
+        expect(response).to_not render_template(:new)
+      end
+    end
```
Comment (TedLivist) [inline spec/requests/posts_spec.rb:47]: - [ ] Kindly move this `describe`  block out of the initial one.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @zieeco,

Good job so far!
✔️ Good documentation
✔️ Tests are passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (zieeco) [reply spec/requests/users_spec.rb:42]: @TedLivist Okay boss

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @zieeco,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**